### ISSUE TITLE
Training on a single video of a person talking

### ISSUE THREAD

**Issue by gfhcs** [VladYushchenko/MDPVideoGeneration#1]
Hello,

we are trying to train the architecture on a single video (around 10k frames), of 1 person talking to the camera. The samples we see, however, do not even show the resemblance of a person. They are just reddish "fog". Is this a sign that we are doing something fundamentally wrong? Or do we just continue training? Is there maybe a technical limitation that makes the technique unsuitable for such kinds of training sets?

Best,
Gereon

**Comment by vladyushchenko** [comment]
Hi, 

thanks for your interest in the repo.

Considering the issue - could you please share the command how you start training and also the dataset (if possible).
Btw, do you use your own data loader or existing one from the repo?

Update: weird images such as reddish "fog" are fine at the very beginning of training. You shall see how generated images gradually become better.

Best,
Vlad

**Comment by prraoo** [comment]
Hello Vlad,

Thank you for your reply, I am @gfhcs teammate and here are the details you asked for:

Here is the command we use for our training using the script `train.py`:
```
checkpoint='', 
dataset='/work/MDPVideoGeneration/mdp_video/dataset/fox/',
deterministic=False, 
dim_z_category=0, dim_z_content=50, dim_z_motion=10, disc_learning_rate=0.0002,
discriminators=['PatchImageDiscriminator', 'PatchVideoDiscriminator'],
every_nth=2, gen_learning_rate=0.0002, generator='VideoGenerator', 
image_batch=32, image_size=64, 
log_folder='work/MDPVideoGeneration/logs/fox2disc//_nocat/Baseline_noise_deterministic/Seed_0/TrainLength_16/Batch_32',
n_channels=3, n_iterations=100000, n_temp_blocks=3, noise_sigma=0.0, num_workers=4, optimizer='adam', print_every=100, save_every=1000, 
save_graph=False, seed=0, show_data_interval=500, snapshot_init=False, temporal_beta=1.0, temporal_kernel_size=3, temporal_sigma=1.0, trainer_type='vanilla_gan', 
use_infogan=False, use_noise=False, video_batch=32, video_length=16
```

Our Dataset Structure is as follows:
Where we have a person talking and we label it as class 0 (single category) and the subdirectories have frames extracted from the video. It can be summarised in a tree structure as below:
```
 person
│   └── 0
│       ├── 0
│       ├── 1
│       ├── 2
│       ├── 3
...
```
Also, we just use the existing data loader and training scripts. Unfortunately, the reddish fog remains throughout the training.
So it would be really helpful if you could let us provide some insights to improve the training.

**Comment by vladyushchenko** [comment]
Hey, 

thanks for the info. The data structure seems to be fine. The UCF-101 dataset structure is basically the following:
```
ucf_train
└───class_0
│   └───video_0
│       │   frame_0.png
│       │   frame_1.png
│       │   ...
│     ...
└───class_1
 |     └───video_0
│       │   frame_0.png
│       │   frame_1.png
│       │   ...    
 |        ....
...
```
Well, path to the dataset is an absolute path to the ucf_train folder, which in your case also seems to be fine.

I will try to run training now on UCF-101 and see if the training correctly proceeds. 
The repo and docs were heavily refactored, therefore some some things may be not up-to-date. Will update you asap ;)

**Comment by prraoo** [comment]
Thanks a lot :)

**Comment by vladyushchenko** [comment]
Hi, 

I have launched the training with the actions dataset from our baseline paper and images were fine.

The dataset you can find here:
[actions_raw_jpg.zip](https://github.com/VladYushchenko/MDPVideoGeneration/files/6164100/actions_raw_jpg.zip)

You can try it out and see the training progress in the Tensorboard logs. 

As for your problem - which video size do you have? We train on images of size 64. Since we based our work on MocoGAN, it can happen that video generator is not powerful enough for your case (it is just standard DCGAN architecture). 

UPDATE:
Besides, I have updated the repo, since there were some package conflicts.

Now the setup should run smoothly. 
There is also an important aspect - now we have 2 training scripts/Pycharm configs that launch training for the mocogan_baseline and one for our temporal_mdp model. Previously I included only training for mocogan baseline - sorry for that.

Here are launcher scripts:
- launcher_template_mocogan_baseline.sh
- launcher_template_temporal_mdp.sh 


Hope it helps you further!

**Comment by prraoo** [comment]
Thank you, I will check the update and adapt the changes in our training.
